app/cache/cache-service.js

```diff
@@ -0,0 +1,32 @@
+const NodeCache = require('node-cache');
+const { Logger } = require('@hmcts/nodejs-logging');
+const logger = Logger.getLogger('CacheService');
+
+class Cache {
+
+  constructor(name, ttlSeconds, checkPeriodSeconds) {
+    this.cache = new NodeCache({ stdTTL: ttlSeconds, checkperiod: checkPeriodSeconds });
+    this.name = name;
+  }
+
+  get(key, storeFunction) {
+    const value = this.cache.get(key);
+    if (value) {
+      return Promise.resolve(value);
+    }
+
+    return storeFunction().then(result => {
+      this.cache.set(key, result);
```

Comment:
the catch section will handle any error coming from the then.(f() ) . However if we have any problem coming from the storeFunction() which happen to be cached-user-resolver.js . that will be lost.   Errors from the implementation (cached-user-resolver.js   and  ./user-resolver or  ) are going to be lost . We could add an small try catch in the cache resolver :

 try{
    return userInfoCache.get
    (jwtUtil.getJwt(jwt),
      () => {
        userResolver.getUserDetails(jwt)
      }
    );
  } catch (e) {
    console.log(`Error in store function for cache aksajbdaihvdaiv i`);
  }

Reply:
FYI: To mimic error during store add `maxKeys: 1` to the NodeCache options then open two browser sessions.

NB: failure during store in matching ccd-case-activity-api implementation is generating a 401 error rather than gracefully falling back to no-caching implementation.  Is this the expected behaviour.